src/formatDate.js

```diff
@@ -50,7 +50,91 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR_INDEX = 3;
+  const YEAR_LIMIT = 30;
+  const CENTURY_21 = 20;
+  const CENTURY_20 = 19;
+
+  const dateStorage = {
+    day: null,
+    month: null,
+    year: null,
+
+    oldFormat: date.split(fromFormat[SEPARATOR_INDEX]),
+    newFormat: [],
+  };
+
+  const { oldFormat, newFormat } = dateStorage;
+  let { day, month, year } = dateStorage;
+
+  for (let i = 0; i < SEPARATOR_INDEX; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = oldFormat[i];
+        break;
+
+      case 'MM':
+        month = oldFormat[i];
+        break;
+
+      case 'YY':
+        year = normalizeYear(oldFormat[i]);
+        break;
+
+      case 'YYYY':
+        year = normalizeYear(oldFormat[i]);
+        break;
+    }
+  }
+
+  for (let i = 0; i < SEPARATOR_INDEX; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
```

Comment:
Would be great to use variables instead of hardcoded values to avoid duplicating